### PR TITLE
[Mobile snaps] Update snaps packages to version 2.0.2

### PR DESCRIPTION
**Pull request by owencraston**
## **Description**

Update snaps controller to version 2.0.2. 

This also involved updating some of the peer dependancies such as `@metamask/slip44`, `@metamask/snaps-utils"`, `@metamask/rpc-methods`. 

Updated the snaps permissions titles to include the newer permissions. The list of permissions can be found [here](https://www.notion.so/bac3299d2c5241c599d2e5e7986e72f7?v=ef742a61bd844435b7171bd2e90b447e).

This changed how snap are installed now...

Before:
- installing a snap would be would trigger a `wallet_installSnap` approval. From this point we would parse the approval request data to show the requested permissions. We would then handle the install state internally.

Now:
- Installing a a snap triggers a `wallet_requestPermissions` approval type. For this we show the `InstallSnapConnectionRequest`. Approving this triggers a second approval request of type `wallet_installSnap`. This is where we show the requested permissions inside the `InstallSnapPermissionsRequest`. Once approved we show a success/error screen

Since the install logic relies so heavily on the the `useApprovalRequest` response, I moved all of the install logic from `app/components/UI/...` into `app/components/Approvals/InstallSnapApproval`

## **Related issues**

Progresses: https://github.com/MetaMask/accounts-planning/issues/116
Fixes: https://github.com/MetaMask/accounts-planning/issues/115
Fixes: https://github.com/MetaMask/accounts-planning/issues/114

## **Manual testing steps**

1. build branch
2. run `yarn setup`
3. navigate to browser tab
4. go to http://metamask.github.io/snaps/test-snaps/1.1.0
5. install BIP-32 snap
6. go through approval flow
7. the snap should be installed
8. navigate to settings/snaps
9. the snap should be available in settings
10. you should be able to preview all of the snaps permissions in the snap settings

## **Screenshots/Recordings**

https://github.com/MetaMask/metamask-mobile/assets/22918444/317d7eca-c0b4-44d1-a60f-e6f1352331c3

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by owencraston** [comment]
@SocketSecurity ignore @metamask/snaps-controllers@2.0.2

**Comment by owencraston** [comment]
@SocketSecurity ignore @metamask/snaps-utils@2.0.1

File: app/core/Engine.ts [hunk 1]
```diff
@@ -623,6 +608,27 @@ class Engine {
       bridge.setupProviderConnection();
     };
 
+    // TODO
```
Comment (Jonathansoufer) [inline app/core/Engine.ts:611]: What's this comment for?

Comment (owencraston) [reply app/core/Engine.ts:611]: We need to make this dynamic, the same way it is [handled on extension](https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/metamask-controller.js#L1127). I will improve this TODO message.

**Review by Jonathansoufer — APPROVED** [review]
LGTM. Good job and well structure code.

**Pull request merged**